### PR TITLE
Dispatch commands for notification messages

### PR DESCRIPTION
**Pull request by akm**
In order to call different commands by notification messages, `blocks-concurrent-subscriber` select a command from commands in configuration file.

## Behavior changes

- The configurations must be in configuration file [example](https://github.com/groovenauts/blocks-concurrent-subscriber/blob/56d349568c62ffcf7bb1159c599d423d31d101b4/test/config1.json)
- The command argument is only `--config`
- Table schema changed [diff](https://github.com/groovenauts/blocks-concurrent-subscriber/commit/a80c13bbaabb7e0291fc57ab4e25c5a35f82b460#diff-f7ac11a1616b17e063ea783346ecd8a3)
- Expects attributes named`completed` and `level` in notification messages

## Other changes

- Use [blocks-variable](https://github.com/groovenauts/blocks-variable) to build command 
- Add a test with mysql database
- Add Makefile for build and test

### CODE REVIEW

**Comment by akm** [comment]
@minimum2scp I did it. Thank you!

File: progress_store.go [hunk 1]
```diff
@@ -31,17 +30,17 @@ func (ss *SqlStore) setup(ctx context.Context, driver, datasource string) (func(
...
 }
 
-func (ss *SqlStore) save(ctx context.Context, pipeline, msg_id string, progress int, publishTime time.Time, f func() error) error {
+func (ss *SqlStore) save(ctx context.Context, pipeline string, msg *Message, f func() error) error {
 	err := ss.transaction(func(tx *sql.Tx) error {
-		_, err := tx.Exec(SQL_UPDATE_JOBS, progress, msg_id, progress)
+		_, err := tx.Exec(SQL_UPDATE_JOBS, msg.progress, msg.msg_id, msg.progress)
 		if err != nil {
-			fmt.Println("Failed to update pipeline_jobs message_id: %v to status: %v cause of %v", msg_id, progress, err)
+			fmt.Println("Failed to update pipeline_jobs job_message_id: %v to progress: %v cause of %v", msg.msg_id, msg.progress, err)
```
Comment (minimum2scp) [inline progress_store.go]: `s/fmt.Println/fmt.Printf/` ?

File: progress_store.go [hunk 2]
```diff
@@ -31,17 +30,17 @@ func (ss *SqlStore) setup(ctx context.Context, driver, datasource string) (func(
...
 			return err
 		}
 
-		_, err = tx.Exec(SQL_INSERT_LOGS, pipeline, msg_id, progress, publishTime)
+		_, err = tx.Exec(SQL_INSERT_LOGS, pipeline, msg.msg_id, msg.publishTime, msg.progress, msg.completedInt(), msg.level, msg.data)
 		if err != nil {
-			fmt.Println("Failed to insert pipeline_job_logs pipeline: %v, message_id: %v, status: %v cause of %v", pipeline, msg_id, progress, err)
+			fmt.Println("Failed to insert pipeline_job_logs pipeline: %v, job_message_id: %v, progress: %v cause of %v", pipeline, msg.msg_id, msg.progress, err)
```
Comment (minimum2scp) [inline progress_store.go]: `s/fmt.Println/fmt.Printf/` ?

File: progress_store.go [hunk 3]
```diff
@@ -53,7 +52,7 @@ func (ss *SqlStore) save(ctx context.Context, pipeline, msg_id string, progress
 		return nil
 	})
 	if err != nil {
-		fmt.Println("Failed to begin a transaction message_id: %v to status: %v cause of %v", msg_id, progress, err)
+		fmt.Println("Failed to begin a transaction job_message_id: %v to progress: %v cause of %v", msg.msg_id, msg.progress, err)
```
Comment (minimum2scp) [inline progress_store.go]: `s/fmt.Println/fmt.Printf/` ?

File: progress_store.go (hunk 1 above)
Comment (akm) [inline progress_store.go]: @minimum2scp Yes. I added `make check` task to find problems like this. And I fixed it.

File: progress_store.go (hunk 2 above)
Comment (akm) [inline progress_store.go]: @minimum2scp I fixed it.

File: progress_store.go (hunk 3 above)
Comment (akm) [inline progress_store.go]: @minimum2scp I fixed it.

**Pull request merged**